### PR TITLE
Also add -pthread to prebuildopts of cryptography

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@Flamefire Can you clarify why this is needed, or what problem is solved with this change?

**Comment by Flamefire** [comment]
Sure: We already do this in the preinstallopts to solve the linking failure on e.g. POWER (I'm sure you remember all the discussions, so won't repeat here)

For older Pythons we don't use pip to install stuff (for likely historic reasons setup.py is the default which is IMO unfortunate) and there the build step is executed while for pip installs only the install step is executed.
Hence for the non-pip builds the preinstallopts are to late as `setup.py build` already needs the `-pthread`.

TLDR: This is the same as we do for `pip install .` but for `setup.py build` which is used by e.g. the 2018 toolchain generation.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @Flamefire!
